Makefile

```diff
@@ -564,6 +566,50 @@ git-submodule:
 	git submodule sync --recursive
 	git submodule update --init --recursive
 
+define PRINT_OPTION
+    ifeq ($$(SUBPROJECT_OPTION),)
+        ALL_OPTIONS += $$(KEYBOARD_OPTION)-$$(KEYMAP_OPTION)
+    else
+        ALL_OPTIONS += $$(KEYBOARD_OPTION)-$$(SUBPROJECT_OPTION)-$$(KEYMAP_OPTION)
+    endif
+endef
+
+define ALL_KEYMAPS
+    KEYMAPS_OPTION := $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(KEYBOARD_OPTION)/keymaps/*/.)))
+    LAYOUTS :=
+    $$(eval -include $(ROOT_DIR)/keyboards/$$(KEYBOARD_OPTION)/rules.mk)
+    KEYBOARD_LAYOUTS := $$(LAYOUTS)
+    ifneq ($$(SUBPROJECT_OPTION),)
+        SP_KEYMAPS_OPTION := $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(KEYBOARD_OPTION)/$$(SUBPROJECT_OPTION)/keymaps/*/.)))
+        KEYMAPS_OPTION := $$(sort $$(KEYMAPS_OPTION) $$(SP_KEYMAPS_OPTION))
+    endif
+
+    LAYOUT_KEYMAPS_OPTION :=
+    $$(foreach LAYOUT_OPTION,$$(KEYBOARD_LAYOUTS),$$(eval LAYOUT_KEYMAPS_OPTION += $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/layouts/*/$$(LAYOUT_OPTION)/*/.)))))
+    
+    KEYMAPS_OPTION := $$(sort $$(KEYMAPS_OPTION) $$(LAYOUT_KEYMAPS_OPTION))
+
+    $$(foreach KEYMAP_OPTION, $$(KEYMAPS_OPTION), $$(eval $$(call PRINT_OPTION)))
+endef
+
+define ALL_SUBPROJECTS
+    SUBPROJECTS_OPTION := $$(notdir $$(patsubst %/rules.mk,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(KEYBOARD_OPTION)/*/rules.mk)))
+    $$(eval $$(call ALL_KEYMAPS))
+    ifneq ($$(SUBPROJECT_OPTION),)
```

Comment:
Found the issue, this line should check for `SUBPROJECTS_OPTION` instead

Reply:
Ah, yeah - thanks!